src/traces/scattergl/index.js

```diff
@@ -47,116 +48,73 @@ function calc(gd, trace) {
     var xa = Axes.getFromId(gd, trace.xaxis);
     var ya = Axes.getFromId(gd, trace.yaxis);
     var subplot = fullLayout._plots[trace.xaxis + trace.yaxis];
+    var count = trace._length;
+    var count2 = count * 2;
     var stash = {};
+    var i, xx, yy;
 
+    var x = xa.makeCalcdata(trace, 'x');
+    var y = ya.makeCalcdata(trace, 'y');
 
-    var x = xaxis.type === 'linear' ? trace.x : xaxis.makeCalcdata(trace, 'x');
-    var y = yaxis.type === 'linear' ? trace.y : yaxis.makeCalcdata(trace, 'y');
-
-    var count = trace._length, i, xx, yy;
+    // we need hi-precision for scatter2d,
+    // regl-scatter2d uses NaNs for bad/missing values
+    //
+    // TODO should this be a Float32Array ??
+    var positions = new Array(count2);
```

Comment:
@dfcreative could we set up `positions` with a `Float32Array` here?

Reply:
Float32 is not enough precision for some timedate and precise linear/etc ranges.